### PR TITLE
suggest format method after error due to old syntax 

### PR DESCRIPTION
**Pull request by jgabry**
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and agree to license (see below)

#### Summary

Advances #849 (doesn't address the interactive part)

Suggests `format` method if compilation error due to old syntax. This doesn't yet do the formatting if in interactive mode, just suggests looking at the example in the doc for `format`. I added an example to that doc showing how to change old syntax. 

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting
(this will be you or your assignee, such as a university or company):
**Columbia University**

By submitting this pull request, the copyright holder is agreeing to
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by jgabry** [comment]
### Example

```r
f <- write_stan_file(
"
parameters {
  real x[3];
}
model {
 x ~ normal(0, 1);
}
"
)
m <- cmdstan_model(f)
```
```
Compiling Stan program...
Error in '/var/folders/s0/zfzm55px2nd2v__zlw5xfj2h0000gn/T/RtmpjGrUuu/model-14ede482935d4.stan', line 3, column 2: Declaration
    of arrays by placing brackets after a variable name was removed in Stan
    2.33.0. Instead use the array keyword before the type. This can be
    changed automatically using the auto-format flag to stanc

make: *** [/var/folders/s0/zfzm55px2nd2v__zlw5xfj2h0000gn/T/RtmpjGrUuu/model-14ede482935d4.hpp] Error 65

Error: An error occured during compilation! See the message above for more information. 
To fix deprecated or removed syntax please see ?cmdstanr::format for an example.
```

The last line of the error message is now. But instead of suggesting _how_ to call the format method I have it point to the example in the format doc that I updated. The user needs to set `compile=FALSE` when calling `cmdstan_model` and then run the `format` method, which is a bit much to describe in the error message. Easier to just see the example in the doc. 

We can also do this interactively in the future, but as a start this message should help.

**Comment by jgabry** [comment]
This might conflict with changes in https://github.com/stan-dev/cmdstanr/pull/836, so we should merge that first.

**Comment by jgabry** [comment]
> This might conflict with changes in #836, so we should merge that first.

Just merged that branch into this one, so I can avoid conflicts

**Comment by jgabry** [comment]
@andrjohns @rok-cesnovar @avehtari This is ready for a look if any of you have a chance. We should wait to merge it until #836 is merged (I'll merge that one if the checks that are running pass).

File: R/model.R [hunk 1]
```diff
@@ -929,6 +934,21 @@ CmdStanModel$set("public", name = "check_syntax", value = check_syntax)
...
+#' file <- write_stan_file("
+#' parameters {
+#'   real x[2];
+#' }
+#' model {
+#'   x ~ std_normal();
+#' }
+#' ")
+#' mod <- cmdstan_model(file, compile = FALSE)
+#' mod$format(canonicalize = TRUE)
```
Comment (avehtari) [inline R/model.R:949]: I would strongly prefer the example for fixing old syntax to be
```
mod$format(canonicalize = list("deprecations"))
```
to keep the changes and possible git diffs minimal, when people want to just update the deprecated syntax.

I also suggest showing
```
mod$format(canonicalize = list("deprecations"), overwrite_file=TRUE)
mod$compile()
```

**Review by avehtari — COMMENTED** [review]
The proposed error message is fine, I'd make changes to the format doc

File: R/model.R (hunk 1 above)
Comment (jgabry) [reply R/model.R:949]: Thanks, sounds good. I'll make the updates.

**Comment by jgabry** [comment]
I updated the doc based on @avehtari's suggestion. Will merge if everything passes.

**Pull request merged**